sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/joins.scala

```diff
@@ -484,6 +484,26 @@ trait JoinSelectionHelper {
     hint.rightHint.exists(_.strategy.contains(NO_BROADCAST_AND_REPLICATION))
   }
 
+  def canLeftSideBeBroadcast(
```

Comment:
This is only called once in `InjectRuntimeFilter`, shall we inline it?

Reply:
+1

Reply:
done